akka-actor/src/main/scala/akka/util/ByteString.scala

```diff
@@ -274,8 +286,44 @@ object ByteString {
     def asByteBuffers: scala.collection.immutable.Iterable[ByteBuffer] = bytestrings map { _.asByteBuffer }
 
     def decodeString(charset: String): String = compact.decodeString(charset)
+
+    private[akka] def writeToOutputStream(os: OutputStream): Unit =
+      bytestrings.foreach(_.writeToOutputStream(os))
```

Comment:
I see that there are possibly some performance differences related to allocation with a ByteString being serialized vs not (slicing over time, etc).... thats a good question. That also relates back into other 'sparseness' concerns in the existing ByteString. For example large IndexedSeqs are copied as flat Arrays via the tmp buffer, when it may be advantageous to spare out all incoming data via the ByteStringBuilder for the same reasons (I believe the Protobuf ByteString does this also).